include/xtensor/xmath.hpp

```diff
@@ -1817,36 +1817,42 @@ XTENSOR_INT_SPECIALIZATION_IMPL(FUNC_NAME, RETURN_VAL, unsigned long long);
      * @param axes the axes along which the mean is computed (optional)
      * @return an \ref xexpression
      */
-    template <class E, class X>
+    template <class T = void, class E, class X>
     inline auto mean(E&& e, X&& axes)
     {
+        using value_type = typename std::conditional_t<std::is_same<T, void>::value, double, T>;
```

Comment:
`T` could be defaulted to `double` (same principle as previous remark)

Reply:
(Just responding to this one but applies for all) In this case, it is fine, but in the general case of sum, you cannot do that because `E` is not yet defined since the template searches for types from left to right. I'll update a PR with the appropriate changes

Reply:
Ah indeed, I hadn't noticed that `RESULT_TYPE` was expanded to `typename std::decay_t<E>::value_type` in the previous case. You can therefore ignore my previous remark.